[집터뷰 작성 - 05] 두 번째 스텝 화면 구현

HayleyKim0716 opened the pull request:
### 1. 개요
집터뷰 작성 단계 중 두 번째 스텝 화면 구현

### 2. 작업사항
* 이전 화면에서 넘어오는 로직 구현
* UI 구현

### 3. 변경로직
* N/A

### 4. 스크린샷
<img src="https://user-images.githubusercontent.com/43338377/185161583-0200efaf-4400-4d3c-87d9-7673a0698c98.png" width="300">

### 5. 참고링크
- [디자인](https://www.figma.com/file/VBtuKLaxlF1ZaKEXCh505b/hometerview?node-id=459%3A18338)

### 6. 기타
* 거주층 선택 반팝업은 다음 PR 로 생성 예정

HayleyKim0716 committed:
#6 fix/집터뷰 작성 화면 : 두 번째 단계 리뷰 작성 화면 UI 구현

HayleyKim0716 committed:
#6 fix/집터뷰 작성 화면 : 두 번째 단계 리뷰 작성 화면 xml 파일명 변경에 따른 binding 타입 변경

HayleyKim0716 committed:
#6 fix/집터뷰 작성 화면 : AndroidManifest.xml 에서 불필요하게 수정된 코드 원복

HayleyKim0716 committed:
Merge branch '6/04_change_names' into 6/05_step2_review

likppi10 commented on app/src/main/java/com/ftw/hometerview/ui/bindingadapter/RatingBarBindingAdapter.kt:
```diff
@@ -0,0 +1,23 @@
+import androidx.databinding.InverseBindingListener
+
+@BindingAdapter("android:rating")
+fun setRating(ratingBar: RatingBar, rating: Float?) {
```
rating이 없을 때는 0으로 설정되지 않나요오..? 그러면 nullable하지 않아도 되려나요?

HayleyKim0716 replied:
두 번째 parameter 가 null 로 들어오는 경우가 있고, 이 때 exception 이 발생해서 nullable 하게 선언해야 한다고 들었는데,
nullable 하지 않게 선언하더라도 정상동작 하네 🤔
exception 이 발생하거나 정상동작하지 않는 경우 모두 겪어봤군,, nullable 하지 않게 선언하고 사용하면서 좀 더 찾아볼게! 고마와!

likppi10 commented on app/src/main/java/com/ftw/hometerview/ui/review/CreateReviewActivity.kt:
```diff
@@ -11,29 +11,51 @@ import com.ftw.hometerview.extension.addFragment
-        addFragment(R.id.fragment_container_view, CreateReviewFirstStepInputAddressFragment.newInstance(), true)
+    override fun onClickNextFromFirstStepResidentialFloor(address: String, floor: String) {
+        addFragment(
+            R.id.fragment_container_view,
```
이펙티브 코틀린에서 봤는데, 뭔가 이렇게 줄바꿈 하는 것에 단순히 줄바꿈으로 가시성을 확보한다 말고도, 
그 누나 xml /> 줄바꿈 처리하는 것처럼 
파라미터가 그렇게 많은 함수면 파라미터의 변동도 있을 건데, 이렇게 줄바꿈을 해주면 추가된 파라미터만 PR에서 확인할 수 있다는게 또 좋은것 같아요!

HayleyKim0716 replied:
ㅋㅋㅋ맞아! 전체 코드가 길어지는 것보다 코드 보기도 쉽고 변경사항 파악도 쉽다는 장점이 크다고 봐! 🙂

likppi10 reviewed (approved):
고생이 많으십니당 :D

HayleyKim0716 committed:
#6 fix/홈 화면 : 불필요한 BindingAdapter 제거
- 중복된 RatingBar 관련 BindingAdapter 제거
- RatingBar BindingAdapter 의 parameter 를 non-nullable 형태로 변경
